crates/starknet-core-contract-client/src/interfaces/messaging.rs

```diff
@@ -100,10 +110,12 @@ where
         selector: U256,
         payload: Vec<U256>,
         nonce: U256,
-    ) -> Result<MessageHash, Error<M>> {
+    ) -> Result<Option<TransactionReceipt>, Error<M>> {
         self.as_ref()
             .start_l1_to_l2_message_cancellation(to_address, selector, payload, nonce)
-            .call()
+            .send()
+            .await
+            .map_err(Into::<ContractError<M>>::into)?
```

Comment:
I thought `call` was the preferred method to interact with eth smart contracts.
When should we use `call` and when should we use `send`?

Reply:
send is for writes and call for reads

Reply:
oookay. Mb. Thanks for fixing it everywhere